engines/registry/lib/registry/engine.rb

```diff
@@ -1,6 +1,32 @@
+require 'fileutils'
+
 module Registry
   class Engine < ::Rails::Engine
     isolate_namespace Registry
     config.generators.api_only = true
+
+    config.after_initialize do
+      Api::Connect::V3::Systems::ActivationsController.class_eval do
+        before_action :handle_cache, only: %w[index]
+
+        def handle_cache
+          # get request header
+          if request.headers['X-Refresh-Registry-Creds'] && ZypperAuth.verify_instance(request, logger, @system)
```

Comment:
The header name can be changed

The main part is to validate the instance metadata, notice that the validation of the instance metadata would put it in the RMT cache if valid

Reply:
That header is sent to that endpoint in every use, I used a different one to make the verification explicit for this use case, meaning not instance verification if not registry call

Using `X-Instance-Header` is fine but it would mean we would trigger an instance verification and a potential update for non registry calls

Reply:
What I understand from my research for https://github.com/SUSE/rmt/pull/1130 suseconnect never sends that header. Or am I wrong there? 
I think it's fair to validate the instance data when it is send and the client is not in the instance data cache.

Reply:
the [client](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L421-L426)  to `connect/systems/activations` route
 
 > it's fair to validate the instance data when it is send
 
 I agree, I will change the header and will work on adjusting the tests for this